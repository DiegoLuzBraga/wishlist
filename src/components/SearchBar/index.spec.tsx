import { render, fireEvent, cleanup, screen } from "@testing-library/react";
import { SearchBar } from ".";

let searchQuery = "";
const onChange = (value: string) => (searchQuery = value);

describe("Seachbar tests", () => {
  it("should match snapshots", () => {
    const { container } = render(
      <SearchBar searchQuery={searchQuery} setSearchQuery={onChange} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should change values", () => {
    const { getByPlaceholderText } = render(
      <SearchBar searchQuery={searchQuery} setSearchQuery={onChange} />
    );

    const input = getByPlaceholderText("Buscar") as HTMLInputElement;

    fireEvent.change(input, { target: { value: "Camisa" } });

    cleanup();

    render(
      <SearchBar searchQuery={searchQuery} setSearchQuery={onChange} />
    );

    expect(searchQuery).toBe("Camisa");
    expect(screen.getByDisplayValue("Camisa")).toBeInTheDocument();
  });
});
