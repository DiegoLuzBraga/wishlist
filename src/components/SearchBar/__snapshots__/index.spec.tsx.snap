// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seachbar tests should match snapshots 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root css-1h6z0sw-MuiFormControl-root-MuiTextField-root"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1w31g1s-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r0:"
        placeholder="Buscar"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-ihdtdm"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
